### PR TITLE
Add more docs

### PR DESCRIPTION
**Pull request by jekiapp**
Add more docs how to use Do() Return() DoAndReturn(). Also add example case how to do custom Matcher.

### CODE REVIEW

**Comment by ahmadmuzakki** [comment]
I signed it!

File: gomock/controller.go [hunk 1]
```diff
@@ -51,8 +51,99 @@
 //         mockObj.EXPECT().SomeMethod(3, "third"),
 //     )
 //
-// TODO:
```
Comment (ariefrahmansyah) [inline gomock/controller.go]: It's better to keep the TODO.

Comment (ahmadmuzakki) [reply gomock/controller.go]: what do you thing TODO next?

**Comment by poy** [comment]
@jekiapp Thanks for the PR.

I was looking at these doc changes via `godoc --http ':9999' and noticed the formatting looked a little off. Do you mind fixing that?

**Comment by jekiapp** [comment]
sure, I'll find some times this week

**Comment by marco-m** [comment]
@jekiapp ping? :-)

**Comment by jekiapp** [comment]
oh god, thanks for pinging me @marco-m. checking it right now :smile:

File: gomock/controller.go [hunk 2]
```diff
@@ -51,8 +51,99 @@
...
+//    }
+//
+// In some cases we can't be sure the exact param would be passed into the function we want to mock.
+// For this case we need to implement our own logic to check if the mock is valid.
+// Suppose we have code like this:
+//    type Metric interface {
+//        PushMetric(duration float64)
+//    }
+//
+//    var metrics Metric
```
Comment (poy) [inline gomock/controller.go]: nit: Can we not use a global variable here?

File: gomock/controller.go [hunk 3]
```diff
@@ -51,8 +51,99 @@
...
+//        mockObj.EXPECT().DoSomething().DoAndReturn(func()error{
+//             fmt.Println("Doing something")
+//             return fmt.Errorf("Error on doing something")
+//        })
+//
+//        err := mockObj.DoSomething() // print Doing something
+//        fmt.Println(err) // print Error on doing something
+//    }
+//
+// In some cases we can't be sure the exact param would be passed into the function we want to mock.
```
Comment (poy) [inline gomock/controller.go]: I'm a little confused by this. It seems like you're giving an example on custom matcher more than anything else. Can you highlight that instead?

File: gomock/controller.go [hunk 4]
```diff
@@ -51,8 +51,99 @@
...
+//    var metrics Metric
+//
+//    func DoSomething(){
+//        start := time.Now()
+//        // do something
+//        duration := time.Since(start)
+//        metrics.PushMetric(duration.Seconds())
+//    }
+//
+// And here's the test.
```
Comment (poy) [inline gomock/controller.go]: nit: Should the period (`.`) be a colon (`:`) to match the other formatting?

**Comment by codyoss** [comment]
@jekiapp still interested in these updates?

**Comment by codyoss** [comment]
I am closing this for now due to lack of response. Please reopen if you are interested in making the updates.

**Pull request closed without merging**